src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -57,9 +57,23 @@ public class Continuation {
 
     /** Reason for pinning */
     public enum Pinned {
-        /** Native frame on stack */ NATIVE,
-        /** Monitor held */          MONITOR,
-        /** In critical section */   CRITICAL_SECTION }
+        NATIVE(2, "Native frame or <clinit> on stack"),
+        MONITOR(3, "Monitor held"),
+        CRITICAL_SECTION(4, "In critical section");
```

Comment:
Is there a reason that the `reasonCode` values does not match the `freeze_result` reason values used in `pinnedReason(int reason)` to create one of these? 

I cannot see that it is used either. Only seem to be read for JFR VirtualThreadPinned Event which only uses the string.

Reply:
These have been updated with the latest JFR changes.